CalendarPicker/index.js

```diff
@@ -65,8 +65,15 @@ export default class CalendarPicker extends Component {
       doStateUpdate = true;
     }
 
+    if (this.props.selectedStartDate && this.props.selectedEndDate &&
+      !moment(prevState.selectedStartDate).isSame(this.props.selectedStartDate, 'day') &&
+      !moment(prevState.selectedEndDate).isSame(this.props.selectedEndDate, 'day')) {
+      doStateUpdate = true;
+    }
```

Comment:
The `doStateUpdate` flag is a strategy to queue up all the state changes and call setState() once at the end instead of for every single condition.  A new empty object should be created and populated in this condition.  E.g.
```
let selectedDatesState = {}

...
if (...) {
     selectedDatesState = {selectedStartDate, selectedEndDate};
     doStateUpdate = true;
}
```

Then below in the `doStateUpdate` conditional check, include the new object as spread parameter.
```
      this.setState({...newStyles, ...newMonthYear, ...selectedDatesState });
```

Reply:
fixed  (few changes to conditions since was not completely correct) and removed some unused props

Reply:
have no permissions to travis log but local tests runs fine